packages/ui-components/src/display/natValue/stringifyNat.js

```diff
@@ -4,21 +4,39 @@ import { roundToDecimalPlaces } from './helpers/roundToDecimalPlaces.js';
 const CONVENTIONAL_DECIMAL_PLACES = 2;
 const MAX_DECIMAL_PLACES = 100;
 
+const calcTrailingZeros = value => {
+  if (!value) return 0;
+
+  let zeroes = 0;
+  while (value > 0 && value % 10n === 0n) {
```

Comment:
Actually I'm not sure there is, this is technically O(log n) where n is the value of the number. If we have the prime factorization, we can count the number of 2s and 5s which is O(1) time (the number of 2 & 5 pairs we can make from the factors is the number of trailing zeros), but looks like prime factorization takes O(sqrt n) which is slower.

Maybe stringifying and going from there is quicker by a constant factor though.

Reply:
Actually maybe we could do a binary search across powers of 10, which would be O(log log n)

Reply:
Neat ideas! Yeah, I was thinking more in the "constant factor" space since n is so small. Doing anything more here seems like premature optimization to me. I should have been more clear about that in my comment: "I suspect there's a better way but I didn't think of one in a few minutes and I don't think it will make enough difference to put more effort into it"

Reply:
Yea of course not worth the trouble, I was just too tempted to think about it some more